src/target/windows/mod.rs

```diff
@@ -0,0 +1,57 @@
+use std::mem;
+use std::ptr;
+use winapi::shared::minwindef::FALSE;
+use winapi::um::dbghelp;
+use winapi::um::processthreadsapi::{CreateProcessW, PROCESS_INFORMATION, STARTUPINFOW};
+use winapi::um::winbase;
+
+/// This structure holds the state of the debuggee on windows systems
+pub struct Target {
+    proc_info: PROCESS_INFORMATION,
+}
+
+macro_rules! wide_string {
+    ($string:expr) => {{
+        use std::os::windows::ffi::OsStrExt;
+        let input = std::ffi::OsStr::new($string);
+        let vec: Vec<u16> = input.encode_wide().chain(Some(0)).collect();
+        vec
+    }};
+}
+
+impl Target {
+    /// Launch a new debuggee process
+    pub fn launch(path: &str) -> Result<Target, Box<dyn std::error::Error>> {
+        let mut si: STARTUPINFOW = unsafe { mem::zeroed() };
+        let mut pi: PROCESS_INFORMATION = unsafe { mem::zeroed() };
+
+        if unsafe {
+            CreateProcessW(
+                std::ptr::null_mut(),
+                wide_string!(&path).as_mut_ptr(),
+                std::ptr::null_mut(),
+                std::ptr::null_mut(),
+                FALSE,
+                winbase::DEBUG_ONLY_THIS_PROCESS,
+                std::ptr::null_mut(),
+                std::ptr::null_mut(),
+                &mut si,
+                &mut pi,
+            )
+        } == FALSE
+        {
+            return Err(Box::new(std::io::Error::last_os_error()));
+        }
+        // Initialize the SymbolHandler
+        unsafe {
+            dbghelp::SymSetOptions(
+                dbghelp::SymGetOptions() | dbghelp::SYMOPT_DEBUG | dbghelp::SYMOPT_LOAD_LINES,
+            )
+        };
+        if unsafe { dbghelp::SymInitializeW(pi.hProcess, ptr::null(), FALSE) } == FALSE {
```

Comment:
https://docs.microsoft.com/en-us/windows/win32/api/dbghelp/nf-dbghelp-syminitialize

> A process that calls SymInitialize should not call it again unless it calls SymCleanup first. If the call to SymInitialize set fInvadeProcess to TRUE and you simply need to reload the module list, use the SymRefreshModuleList function.

> All DbgHelp functions, such as this one, are single threaded. Therefore, calls from more than one thread to this function will likely result in unexpected behavior or memory corruption. To avoid this, call SymInitialize only when your process starts and SymCleanup only when your process ends. It is not necessary for each thread in the process to call these functions.

This isn't safe to do. You would need to make `Target` neither `Send` nor `Sync`. You also need a static to ensure that `Target` is dropped before a new one is created and call `SymCleanup` from the drop implementation of `Target`.

I think using https://github.com/willglynn/pdb as suggested by @nbaksalyar is better.

Reply:
Thanks @bjorn3  
Yeah I did initially think i would have to use the `pdb` crate but then cancelled out that plan once i read the msdn docs and thought "Winapi has an inbuilt symbol loader. Why not use that."